tests/lib/string_conv/src/main.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/types.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <ztest.h>
+#include <string_conv.h>
+
+struct input {
+	char *str;
+	union {
+		long l_val;
+		unsigned long ul_val;
+		struct {
+			double val;
+			double eps;
+		} dbl_val;
+	} exp_val;
+	bool is_valid;
+};
+
+struct obj_array {
+	struct input elems[20];
+	size_t elem_cnt;
+};
+
+static void test_str2long(void)
+{
+	long test_val = 0;
+
+	const struct obj_array ctx = {
+		.elems = {
```

Comment:
Why not just do `struct input elems[] = { ... };` and then user `ARRAY_SIZE(elems)` instead of `elem_cnt`?